controllers/listingController.js

```diff
@@ -0,0 +1,96 @@
+import Listing from "../models/listingSchema.js";
+import Review from "../models/reviewSchema.js";
+
+const listingIndexPage = async (req, res) => {
+    const allListings = await Listing.find({});
+    res.render("listings/index.ejs", { allListings });
+};
+
+const showNewListing = (req, res) => {
+    res.render("./listings/new.ejs");
+}
+
+const showListing = async (req, res) => {
+    let { id } = req.params;
+    const listing = await Listing.findById(id)
+        .populate({
+            path: "reviews",
+            populate: {
+                path: "author",
+            },
+        })
+        .populate("owner");
+    if (!listing) {
+        req.flash("error", "Listing doesn't exist anymore 🤷🏻‍♀️");
+        res.redirect("/listings");
+    }
+    res.render("listings/show.ejs", { listing });
+}
+
+const createNewListing = async (req, res, next) => {
+    let url = req.file.path;
+    let filename = req.file.filename;
+    console.log(url, "-------", filename);
+    const newListing = new Listing(req.body.listing);
+    newListing.owner = req.user._id;
+    newListing.image = { url, filename };
+    await newListing.save();
+    req.flash("success", "New listing added successfully ✅");
```

Comment:
for now these are okay but if you want to write more better code, you can create enums for status... and constants for FlashMessages and use that everywhere.
So basicallay you will have a constants folders which will contain a file FlashMessages.js that file will look something like...

`export const NEW_LISTING_SUCCESS_MESSAGE = "New listing added successfully ✅"` and all other messages as well.

For enum you can create a enum folder and create a enum file there

Reply:
will do